airbyte-cdk/python/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py

```diff
@@ -166,6 +167,7 @@ def _get_request_options(
         :return:
         """
 
+        # FIXME we should eventually remove the usage of stream_state as part of the interpolation
```

Comment:
I'm not exactly sure how we can deprecate this but I would probably just remove this once we have an alternative for source-intercom (I'm not sure I understand the role of stream_state interpolation here) and source-railz (semi-incremental)

Reply:
I think there's a path towards removing references to the stream_state from source-intercom. [Documented here](https://github.com/airbytehq/airbyte/issues/27641).

Similarly for Rails, I suspect we'll get the filtering we want by filtering on the stream_slice instead of the stream_state.

Reply:
Actually, I think the semi-incremental initiative should solve the case of Railz